scripts/build/build_libpq.sh

```diff
@@ -38,39 +56,101 @@ case "$ID" in
         apk add --no-cache zlib-dev krb5-dev linux-pam-dev openldap-dev openssl-dev
         ;;
 
+    macos)
+        brew install automake m4 libtool
+        # If available, libpq seemingly insists on linking against homebrew's
+        # openssl no matter what so remove it. Since homebrew's curl depends on
+        # it, force use of system curl.
+        brew uninstall --force --ignore-dependencies openssl gettext curl
+        if [ -z "$MACOSX_ARCHITECTURE" ]; then
```

Comment:
I missed that the `-z "$variable"` pattern breaks with bash's *error on undefined variable* mode on. i.e. I've broken non-cross compile mode.

```suggestion
        if [ -z "${MACOSX_ARCHITECTURE-}" ]; then
```
Likewise with the `openssl` check on line 133.

Shall I `commit --amend` the fix into the offending commit. I'm slightly wary of doing a force push whilst we're both pushing to the same branch.

Reply:
Incidentally, are you happy with the name `MACOSX_ARCHITECTURE`? It's sort of the standard variable but not even standard enough that any of these packages' build systems supported it so we could go for something more memorable?

Reply:
I actually thought that it was a standard variable, re-defined for an excess of prudence, but yes, it would break with `-u`. Will fix it (usually I use the `:-` suffix).

Reply:
Thanks.

> usually I use the :- suffix

One day I won't have to look it up every time I need to remember what all the variable expansion rules are...

Reply:
re-running [a new job](https://github.com/psycopg/psycopg2/actions/runs/12622882792) after clearing the runners' caches.